lib/algolia/search_index.rb

```diff
@@ -1045,47 +1045,6 @@ def check_array(object)
         raise AlgoliaError, 'argument must be an array of objects' unless object.is_a?(Array)
       end
 
-      # Check the passed object
-      #
-      # @param object [Object]
-      # @param in_array [Boolean] whether the object is an array or not
-      #
-      def check_object(object, in_array = false)
```

Comment:
not sure what this is, but the change doesn't look related

Reply:
it's kind of related as part of those methods are used in the search client as well, and started to be needed for this feature. But I can make a different PR

Reply:
yes, I think separate PR will be more straightforward :)

Reply:
it's done and merged!